### ISSUE TITLE
[incubator/elasticsearch] chart does not support new xpack syntax

### ISSUE THREAD

**Issue by aba182** [helm/charts#7901]
**Is this a request for help?**:

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG REPORT

**Which chart**: [incubator/elasticsearch]

**What happened**: in elasticsearch 6.3+ xpack is installed by default and the config parameters to enable changed. 

**What you expected to happen**:  We need elasticsearch to allow the enabling of xpack systems

**How to reproduce it** (as minimally and precisely as possible): set `cluster.xpackEnable true`

**Anything else we need to know**: I am testing out fixes and will submit a PR once my testing is completed.

**Comment by iboware** [comment]
I cannot get pods running when I set `cluster.xpackEnable` true, is this the same problem?

`unknown setting [xpack.security.enabled] please check that any required plugins are installed, or check the breaking changes documentation for removed settings at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:140) ~[elasticsearch-6.4.1.jar:6.4.1] `

**Comment by aba182** [comment]
Hey iboware ya it is related. I am just about to submit a PR to fix this with required docs.

**Comment by catmeme** [comment]
I added a note in the code review post the merge, repeating it here for visibility:

`xpack.monitoring.enabled` was not deprecated, and it's required for monitoring to be enabled. This PR has helped us export to csv with xpack, but we cannot enable monitoring without `xpack.monitoring.enabled`.

We modified the configmap directly after upgrading to enable this as a work-around.

https://www.elastic.co/guide/en/elasticsearch/reference/6.4/monitoring-settings.html
